File: Hard/2444. Count Subarrays With Fixed Bounds.java
class Solution {
    public long countSubarrays(int[] nums, int minK, int maxK) {
        var cnt = 0L;
        int maxi = -1, mini = -1;
        int n = nums.length;
        for (int l = 0, r = 0; r < n; r++) {
            if (nums[r] < minK || nums[r] > maxK) {
                l = r + 1;
                continue;
            }
            if (nums[r] == maxK) {
                maxi = r;
            }
            if (nums[r] == minK) {
                mini = r;
            }
            cnt += Math.max(Math.min(maxi, mini) - l + 1, 0);
        }
        return cnt;
    }
}

File: Hard/2444. Count Subarrays With Fixed Bounds.py
from typing import List


class Solution:
    def countSubarrays(self, nums: List[int], minK: int, maxK: int) -> int:
        mini = maxi = badi = -1
        result = 0
        for i, v in enumerate(nums):
            if v < minK or v > maxK:
                badi = i
                continue
            if v == minK:
                mini = i
            if v == maxK:
                maxi = i
            if mini > badi and maxi > badi:
                result += max(min(mini, maxi) - badi, 0)
        return result